src/glum/_distribution.py

```diff
@@ -306,46 +273,41 @@ def deviance_derivative(self, y, mu, sample_weight=1):
 
     def _mu_deviance_derivative(
         self,
-        coef: np.ndarray,
-        X,
-        y: np.ndarray,
-        sample_weight: np.ndarray,
+        coef,
+        X: Union[MatrixBase, StandardizedMatrix],
+        y,
+        sample_weight,
         link: Link,
-        offset: np.ndarray = None,
-    ) -> tuple[np.ndarray, np.ndarray]:
-        """Compute ``mu`` and the derivative of the deviance \
-            with respect to coefficients."""
+        offset=None,
+    ):
+        """Compute ``mu`` and the derivative of the deviance with respect to coefficients."""
         lin_pred = _safe_lin_pred(X, coef, offset)
         mu = link.inverse(lin_pred)
         d1 = link.inverse_derivative(lin_pred)
         temp = d1 * self.deviance_derivative(y, mu, sample_weight)
+
         if coef.size == X.shape[1] + 1:
             devp = np.concatenate(([temp.sum()], temp @ X))
         else:
             devp = temp @ X  # same as X.T @ temp
+
         return mu, devp
 
     def eta_mu_deviance(
         self,
         link: Link,
         factor: float,
-        cur_eta: np.ndarray,
-        X_dot_d: np.ndarray,
-        y: np.ndarray,
-        sample_weight: np.ndarray,
-    ):
-        """
-        Compute ``eta``, ``mu`` and the deviance.
-
-        Compute:
-        * the linear predictor, ``eta``, as ``cur_eta + factor * X_dot_d``;
-        * the link-function-transformed prediction, ``mu``;
-        * the deviance.
+        cur_eta,
+        X_dot_d,
+        y,
+        sample_weight,
+    ) -> tuple[np.ndarray, np.ndarray, float]:
+        """Compute ``eta``, ``mu`` and the deviance.
```

Comment:
```suggestion
        """Compute ``eta``, ``mu``, and the deviance.
```

Reply:
The Oxford comma is optional. ;) And superfluous too.